stackrox/stackrox
# ROX-17328: networflows offline v1

## Description

OfflineV1 implementation of the NetworkFlows component. The NetworkFlows pipeline is divide in two; the receiving/storing of NetworkFlows from collector, and the processing/sending of these NetworkFlows. When we switch to offline mode we only stop the second part of the pipeline, meaning we still receive flows from collector and we save them in our internal caches.

## Checklist

If any of these don't apply, please comment below.

## Testing Performed

    1. Deploy ACS in a cluster with `ROX_PREVENT_SENSOR_RESTART_ON_DISCONNECT` set to `true`
    2. Create a nginx deployment and service to connect to it
    3. Scale down central to break the connection with sensor
    4. Check that sensor is in the retry loop
    5. Create a pod (e.g. `talk`) that connects to the nginx deployment (e.g. `kubectl -n test run talk --rm -i --tty --image quay.io/rhacs-demo/netflow -- -connect nginx-service.test.svc.cluster.local:80`)
    6. Scale up central and stop the pod (`talk`) that connects to the nginx
    7. Check that we can see a flow to the nginx deployment in the NetworkGraph in central
